.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: Learn Layer5
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  docker:
+    name: Docker build and push
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master') && success()
+      uses: actions/checkout@master
+      with:
+        fetch-depth: 1
+    - name: Docker login
+      if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master') && success()
+      uses: azure/docker-login@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Docker build & push
+      if: github.event_name != 'pull_request' && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master') && success()
```

Comment:
Nice, so we only push to Docker Hub when making a versioned release, yeah?

Reply:
Actually, I don't mind either way. Should I make it push every time we merge to master?

Reply:
Actually, I don't mind either way. Should I make it push every time we merge to master?

Reply:
@kanishkarj no, this type of control is great. @naturalett can you implement this same logic in Meshery repo?

Reply:
@rav121, fyi

Reply:
@leecalcote Sure but I will wait until this PR will be merged. Just to be on the safe side.